card/lib/card/migration.rb

```diff
@@ -98,50 +100,24 @@ def migration_paths
     Cardio.paths self.class.type
   end
 
-
   # Execute this migration in the named direction
   # copied from ActiveRecord to wrap "up" in "contentendly"
-  def migrate(direction)
-    return unless respond_to?(direction)
-
-    case direction
-    when :up   then announce "migrating"
-    when :down then announce "reverting"
-    end
-
-    time   = nil
-    ActiveRecord::Base.connection_pool.with_connection do |conn|
-      @connection = conn
-      if respond_to?(:change)
-        if direction == :down
-          recorder = CommandRecorder.new(@connection)
-          suppress_messages do
-            @connection = recorder
-            change
-          end
-          @connection = conn
-          time = Benchmark.measure {
-            self.revert {
-              recorder.inverse.each do |cmd, args|
-                send(cmd, *args)
-              end
-            }
-          }
-        else
-          time = Benchmark.measure { change }
-        end
+  def exec_migration(conn, direction)
+    @connection = conn
+    if respond_to?(:change)
+      if direction == :down
+        revert { change }
       else
-        time = Benchmark.measure { contentedly { send(direction) } }
+        change
       end
-      @connection = nil
-    end
-
-    case direction
-    when :up   then announce "migrated (%.4fs)" % time.real; write
-    when :down then announce "reverted (%.4fs)" % time.real; write
+    else
+      contentedly { send(direction) }
     end
+  ensure
+    @connection = nil
   end
```

Comment:
What do we actually change?  I'm guessing you have copied/customized a rails method.  Nice that the new method is way simpler.